pallets/pallet-asset-switch/src/lib.rs

```diff
@@ -489,11 +500,19 @@ pub mod pallet {
 				return Err(DispatchError::from(Error::<T, I>::Internal));
 			}
 
-			// 10. Send XCM out
-			T::XcmRouter::deliver(xcm_ticket.0).map_err(|e| {
-				log::info!("Failed to deliver ticket with error {:?}", e);
-				DispatchError::from(Error::<T, I>::Xcm)
-			})?;
+			// 10. Send XCM out (only when not benchmarking, as delivery fees are anyway
+			//     accounted for by the router)
+			cfg_if::cfg_if! {
+				if #[cfg(not(feature = "runtime-benchmarks"))] {
+					T::XcmRouter::deliver(xcm_ticket.0).map_err(|e| {
+						log::info!("Failed to deliver ticket with error {:?}", e);
+						DispatchError::from(Error::<T, I>::Xcm)
+					})?;
+				} else {
+					log::trace!(target: LOG_TARGET, "Running benchmarks. Message will not be delivered to destination.");
+					drop(xcm_ticket);
+				}
```

Comment:
I think we can improve that. 
Is it not possible to mock a channel with `ParachainSystem::open_outbound_hrmp_channel_for_benchmarks_or_tests`? A reachable destination can be configured in `BenchmarkHelper`. The destination is then set in the storage map of the pallet. 

Therefore, we don't need a special benchmark logic in the pallet and the calculated weights are more accurate

Reply:
Benchmarks seem to pass after the refactoring, thanks a lot! https://github.com/KILTprotocol/kilt-node/pull/671/commits/2ae571ee40d522c2ed1f05962ab85f394bc88ab0